internal/controlplane/handlers_authz.go

```diff
@@ -334,18 +463,27 @@ func (s *Server) AssignRole(ctx context.Context, req *minder.AssignRoleRequest)
 		return nil, status.Errorf(codes.Internal, "error getting user: %v", err)
 	}
 
-	// Determine target project.
-	entityCtx := engine.EntityFromContext(ctx)
-	projectID := entityCtx.Project.ID
+	// Check in case there's an existing role assignment for the user
+	as, err := s.authzClient.AssignmentsToProject(ctx, targetPrj)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "error getting role assignments: %v", err)
+	}
+
+	for _, a := range as {
+		if a.Subject == identity.String() {
```

Comment:
I think this should also be checking for the role? We should match user + role before calling out that the assignment already exists

Reply:
It was decided that a user can have only one role within a project so in this case this should be enough 👍